### DISCUSSION TITLE
Solution for kodi crash after moving Netflix to a new installation

### DISCUSSION THREAD

**Discussion by matejdro** [CastagnaIT/plugin.video.netflix#1375] [General]
For couple of times I've been experiencing annoying issue when moving Netflix installation from one Kodi installation to another (e.g. reinstalling Android app while keeping the profile folder). When attempting to play videos, Kodi crashes with the following errors:

```
06-18 16:42:53.219 10473 10473 F DEBUG   : Abort message: 'JNI DETECTED ERROR IN APPLICATION: JNI GetByteArrayElements called with pending exception android.media.MediaDrm$MediaDrmStateException: Failed to encrypt: General DRM error
06-18 16:42:53.219 10473 10473 F DEBUG   :   at byte[] android.media.MediaDrm.encryptNative(android.media.MediaDrm, byte[], byte[], byte[], byte[]) (MediaDrm.java:-2)
06-18 16:42:53.219 10473 10473 F DEBUG   :   at byte[] android.media.MediaDrm.access$300(android.media.MediaDrm, byte[], byte[], byte[], byte[]) (MediaDrm.java:135)
06-18 16:42:53.219 10473 10473 F DEBUG   :   at byte[] android.media.MediaDrm$CryptoSession.encrypt(byte[], byte[], byte[]) (MediaDrm.java:1781)
```

Solution to the issue is to go to the:

1. Addon settings
2. Expert
3. ESN / Widevine settings
4. Apply changes now

After that Kodi will not crash anymore and videos should play.

Just a piece of knowledge I want to share in case anyone (including future me) needs it. Could it be also added to the [FAQ](https://github.com/CastagnaIT/plugin.video.netflix/wiki/FAQ-%28Errors%29)?

**Comment by CastagnaIT** [comment]
more than a solution this seems a workaround
is useful to understand about the circumstance

> e.g. reinstalling Android app while keeping the profile folder)

it is not clear what exactly what operations you have done with your device
but this practice seems to me a bit unusual
to give a DRM error something has changed in the environment
therefore it would be necessary to investigate further to see if this situation can be managed in some way

**Comment by matejdro** [comment]
I had to do it for two reasons so far:

* I had to factory reset my Nvidia Shield (2019 Tube) due to issues with the new system update. I did not want to setup Kodi again, so I copied profile folder over from old install to new install
* I changed my Kodi install to [Dolby Vision build](https://forum.kodi.tv/showthread.php?tid=360018). Since this custom build has a different signature, old Kodi app had to be uninstalled first.

**Comment by CastagnaIT** [comment]
have you the full kodi log with the error?

**Reply by matejdro** [reply]
I've tried to reproduce it and I can't anymore. Not sure why. Will update this thread if I encounter the issue again.
